reviewers/JSCSChecker.js

```diff
@@ -0,0 +1,268 @@
+      'no-constant-condition': this.SEVERITY_HIGH,
+      'no-dupe-args': this.SEVERITY_HIGH,
+      'no-dupe-keys': this.SEVERITY_HIGH,
+      'no-duplicate-case': this.SEVERITY_HIGH,
+      'no-ex-assign': this.SEVERITY_HIGH,
+      'no-extra-parens': this.SEVERITY_MEDIUM,
+      'no-extra-semi': this.SEVERITY_MEDIUM,
+      'no-func-assign': this.SEVERITY_HIGH,
+      'no-irregular-whitespace': this.SEVERITY_HIGH,
+      'no-sparse-arrays': this.SEVERITY_HIGH,
+      'no-unexpected-multiline': this.SEVERITY_HIGH,
+      'no-unreachable': this.SEVERITY_HIGH,
+      'valid-jsdoc': this.SEVERITY_HIGH,
+      'valid-typeof': this.SEVERITY_HIGH,
+      'accessor-pairs': this.SEVERITY_HIGH,
+      'array-callback-return': this.SEVERITY_HIGH,
+      'block-scoped-var': this.SEVERITY_HIGH,
+      'complexity': [this.SEVERITY_HIGH, { 'max': 4 }],
+      'consistent-return': this.SEVERITY_HIGH,
+      'curly': this.SEVERITY_HIGH,
+      'dot-notation': this.SEVERITY_HIGH,
+      'eqeqeq': this.SEVERITY_HIGH,
+      'no-eval': this.SEVERITY_MEDIUM,
+      'no-lone-blocks': this.SEVERITY_HIGH,
+      'no-loop-func': this.SEVERITY_HIGH,
+      'no-magic-numbers': this.SEVERITY_HIGH,
+      'no-multi-spaces': this.SEVERITY_HIGH,
+      'no-native-reassign': this.SEVERITY_HIGH,
+      'no-new': this.SEVERITY_HIGH,
+      'no-new-func': this.SEVERITY_HIGH,
+      'no-proto': this.SEVERITY_HIGH,
+      'no-redeclare': this.SEVERITY_HIGH,
+      'no-return-assign': this.SEVERITY_HIGH,
+      'no-self-assign': this.SEVERITY_HIGH,
+      'no-self-compare': this.SEVERITY_HIGH,
+      'no-unmodified-loop-condition': this.SEVERITY_HIGH,
+      'no-unused-expressions': this.SEVERITY_HIGH,
+      'no-useless-call': this.SEVERITY_HIGH,
+      'no-useless-concat': this.SEVERITY_HIGH,
+      'no-useless-escape': this.SEVERITY_HIGH,
+      'no-shadow': this.SEVERITY_LOW,
+      'no-shadow-restricted-names': this.SEVERITY_LOW,
+      'array-bracket-spacing': [this.SEVERITY_LOW, 'never'],
+      'block-spacing': this.SEVERITY_LOW,
+      'brace-style': this.SEVERITY_LOW,
+      'camelcase': this.SEVERITY_LOW,
+      'comma-spacing': this.SEVERITY_LOW,
+      'consistent-this': [ this.SEVERITY_LOW, 'me'],
+      'func-names': this.SEVERITY_LOW,
+      'indent': [this.SEVERITY_LOW, 2],
+      'key-spacing': this.SEVERITY_LOW,
+      'keyword-spacing': this.SEVERITY_LOW,
+      'quotes': [this.SEVERITY_LOW, 'single'],
+      'semi': this.SEVERITY_LOW,
+      'semi-spacing': this.SEVERITY_LOW
+    }
+
+    var messages = 
+    
+    console.log( JSON.stringify( messages ) );
+    
+    // Lint the base source.
+    errors.base = linter.verify( baseSource, {
+      rules: rules 
+    });
+    
+    // Lint the head source.
+    errors.head = linter.verify( headSource, {
+      rules: rules 
+    });
+    
+    // Remember this report.
+    var report = {
+      file: to,
+      errors: errors
+    };
+    
+    this.checkedFiles.push( report );
+  },
+  
+  /**
```

Comment:
Missing JSDoc @returns for function.
```javascript
  /**
```
Missing JSDoc parameter type for 'change'.
```javascript
  /**
```
Missing JSDoc parameter type for 'path'.
```javascript
  /**
```
Missing JSDoc parameter type for 'position'.
```javascript
  /**
```
Missing JSDoc parameter type for 'callback'.
```javascript
  /**
```